client/js/genre.js

```diff
@@ -1,11 +1,28 @@
-/** fetches youtube api calls from YoutubeServlet.java
-    and displays on youtube-genre.html
-*/
+/**
+ * fetches youtube api calls from YoutubeServlet.java
+ * and displays on youtube-genre.html
+ */
 
-const genreBlock = document.getElementById('genres');
+const numVideosInput = document.getElementById('numVideos');
+const DEFAULT_NUM_VIDS = 10;
+
+/**
+ * extracts num vids from input, otherwise returns default
+ * helper fn for displayMusicGenre()
+ * @returns {number} num videos user wants to retrieve
+ */
+function getNumVids() {
+  const numVideos = numVideosInput.value;
+  return numVideos || DEFAULT_NUM_VIDS;
+}
+
+/**
+ * fetches genre count hashmap from /api/youtube and updates html
+ */
 async function displayMusicGenre() {
-  const response = await fetch("/api/youtube");
-  console.log(response.status);
+  const genreBlock = document.getElementById('genres');
+  const numVideos = getNumVids();
+  const response = await fetch(`/api/youtube?num_videos=${numVideos}`);
   if (response.status != 200) {
```

Comment:
check if the response is `401`, not just if it's not `200`; all of the response codes have distinct meanings and most of them don't mean "you are not logged in / authorized"